components/event-simulator/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/CarbonEventSimulator.java

```diff
@@ -601,8 +609,8 @@ public void run() {
                         if (bis != null) {
                             bis.close();
                         }
-                        if (dis != null) {
-                            dis.close();
+                        if (br != null) {
+                            br.close();
```

Comment:
Even this is a Simulator, best practice, IO Stream needs to be closed at "finally" block. "bis" and "br".